src/uniprotkb/components/protein-data-views/SubcellularLocationView.tsx

```diff
@@ -25,25 +25,38 @@ const getSubcelluarLocationId = (id: string) => {
   return null;
 };
 
+// 😷, get style rules from the document context in order to
+// copy and inject them in the shadow dom later through the template
+const getStylesContaining = (...selectors: string[]) =>
+  Array.from(document.styleSheets)
+    .flatMap((stylesheet) =>
+      Array.from(stylesheet.rules).map(
+        (rule) =>
+          selectors.some((selector) => rule.cssText.includes(selector)) &&
+          rule.cssText
+      )
+    )
+    .filter(Boolean) as string[];
+
 const SubcellularLocationView: FC<{
   comments?: SubcellularLocationComment[];
   taxonId: OrganismData['taxonId'];
   lineage: OrganismData['lineage'];
 }> = ({ comments, taxonId, lineage }) => {
-  const ref = useRef(null);
+  const ref = useRef<HTMLDivElement>(null);
 
   // TODO: injecting HTML because of the way the web component is implemented. See here for details: https://stackoverflow.com/questions/43836886/failed-to-construct-customelement-error-when-javascript-file-is-placed-in-head
   // TODO: reference colors directly?
   // TODO: the graphic top is an estimate but instead should be based on the size of the entry header
   useEffect(() => {
-    const sls = comments?.flatMap(({ subcellularLocations }) =>
-      subcellularLocations
-        ?.flatMap(({ location }) => getSubcelluarLocationId(location.id))
-        .filter(Boolean)
-        .join(',')
-    );
     if (lineage && taxonId && comments && !isVirus(lineage)) {
-      ref.current.insertAdjacentHTML(
+      const sls = comments?.flatMap(({ subcellularLocations }) =>
+        subcellularLocations
+          ?.flatMap(({ location }) => getSubcelluarLocationId(location.id))
```

Comment:
By the way, there is probably a typo here in the name of the function.
And also, here `.map` would work just fine as it's returning a single item.

Reply:
Good catch on the typo. I removed the inner `flatMap` but kept the outer as otherwise it's an array of array of strings that is being joined. Could work with just the `map` as joining arrays of arrays yields the required comma separated string as `['a','b'].toString() => 'a,b'` but it's not immediately clear to me that this is why it works out so prefer the explicit flatMap.

Reply:
Yeah, I was just mentionning the inner one. I didn't realise skipping the outer one still made it work with some magic joining logic, but I'm with you on keeping the outer one.